src/compiler/utilities.ts

```diff
@@ -6168,7 +6170,16 @@ namespace ts {
                     }
                     (symlinkedDirectories || (symlinkedDirectories = new Map())).set(symlinkPath, real);
                 }
-            }
+            },
+            setSymlinkedDirectoryFromSymlinkedFile(symlink, real) {
+                const [commonResolved, commonOriginal] = guessDirectorySymlink(real, symlink, cwd, getCanonicalFileName) || emptyArray;

```

Comment:
Should you do this lazily to avoid having to do this by default?
Also dont you want to set `symlinkedFiles` ?

Reply:
Is `symlinkedFiles` just a fast path for searching `symlinkedDirectories`? I was just copying what `discoverProbableSymlinks` does. `symlinkedFiles` is not used by module specifier resolution / auto imports, but from the one place that it is used, it seems like it might be beneficial to add to it whenever we can.

Reply:
Also, I’m not sure making this more lazy is worth the complexity. This method only gets called during the initialization/update of an AutoImportProviderProject, and is altogether a small part of the overhead of getting better auto imports. It seems like most users _would_ invoke this eventually.

Reply:
i was just saying because if completions are not invoked, and program get disposed or changes it might be unnecessary but we can do that later if people report on it..